pkg/reconciler/revision/revision.go

```diff
@@ -78,18 +79,44 @@ func (c *Reconciler) reconcileDigest(ctx context.Context, rev *v1.Revision) erro
 		ServiceAccountName: rev.Spec.ServiceAccountName,
 		ImagePullSecrets:   imagePullSecrets,
 	}
-	digest, err := c.resolver.Resolve(rev.Spec.GetContainer().Image,
-		opt, cfgs.Deployment.RegistriesSkippingTagResolving)
-	if err != nil {
-		err = fmt.Errorf("failed to resolve image to digest: %w", err)
-		rev.Status.MarkContainerHealthyFalse(v1.ReasonContainerMissing,
-			v1.RevisionContainerMissingMessage(
-				rev.Spec.GetContainer().Image, err.Error()))
-		return err
+	if rev.Status.ImageDigests == nil {
+		rev.Status.ImageDigests = make(map[string]string, len(rev.Spec.Containers))
 	}
 
-	rev.Status.ImageDigest = digest
+	var digestGrp errgroup.Group
+	type digestData struct {
+		digestValue        string
+		containerName      string
+		isServingContainer bool
+	}
 
+	digests := make(chan digestData, len(rev.Spec.Containers))
+	for _, container := range rev.Spec.Containers {
+		container := container // Standard Go concurrency pattern.
+		digestGrp.Go(func() error {
+			digest, err := c.resolver.Resolve(container.Image,
+				opt, cfgs.Deployment.RegistriesSkippingTagResolving)
+			if err != nil {
+				err = fmt.Errorf("failed to resolve image to digest: %w", err)
+				rev.Status.MarkContainerHealthyFalse(v1.ReasonContainerMissing,
+					v1.RevisionContainerMissingMessage(
+						container.Image, err.Error()))
+				return err
+			}
+			digests <- digestData{digestValue: digest, containerName: container.Name, isServingContainer: len(container.Ports) != 0}
```

Comment:
Let's please make this a multi-line statement like so:

```golang
digests <- digestData{
  digestValue: digest,
  containerName: container.Name,
  isServingContainer: len(container.Ports) != 0 || len(rev.Spec.Containers) == 1,
}
```

**Note:** I'd also propose to move the "if len == 1" check in here, to have the code below be generic and always just key off of `isServingContainer`. That's a minor thing though.